### PR TITLE
NOMRG see what happens when CI does not use conda-forge globally

### PR DESCRIPTION
**Pull request by mathurinm**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
         activate-environment: ${{ env.CONDA_ENV }}
         python-version: 3.8
         # Use miniforge to only get conda-forge as default channel.
-        miniforge-version: latest
+        # miniforge-version: latest
```
Comment (mathurinm) [inline .github/workflows/main.yml:45]: @tomMoral removing this line breaks the build. So it confirms we do not add conda-forge to the channels through the env.yml as we thought we did. I am fixing it, will send a PR

**Comment by tomMoral** [comment]
I removed the addition of the channel here:
https://github.com/benchopt/benchOpt/commit/20ec9c4f14040c91930f63fccb05d92e7dc91189#diff-5d3b678ce3b9b567980fa1221a99bc6c6250c0720ec268b557ddb9f47080510cL111

**Comment by mathurinm** [comment]
I don't get your point. My PR aims at showing that if the user has not configured his conda to use conda-forge, `benchopt install --env .` fails.

**Comment by mathurinm** [comment]
So for me it means that 
```
# Yaml config file for benchopt env.
BENCHOPT_ENV = """
channels:
  - conda-forge
  - nodefaults
dependencies:
  - python=3.8
  - numpy
  - cython
  - compilers
  - pip
  - pip:
    - {benchopt_requirement}
"""
```

sets the channel only when creating the env, not when installing packages later.

**Comment by tomMoral** [comment]
My point was just a pointer to a piece of code I removed but I shouldn't have :)
I think your analysis is correct and I agree that we should add the channel properly so it works on default `conda` envs.

**Comment by mathurinm** [comment]
Sorry I missed the specific line you pointed to. 

Once we merge https://github.com/benchopt/benchOpt/pull/244  we can see if the build succeeds here

**Comment by mathurinm** [comment]
OK it works now. Do you think we should remove miniforge from all benchmarks workflows to better test their installation @tomMoral ?

**Comment by tomMoral** [comment]
No I don't see the point in doing this for all benchmarks.
We should just add a check in the `benchopt` repo to make sure an env created from conda has the proper channels. It would be nice to simply add an extra entry in the CI for this.

**Comment by mathurinm** [comment]
The point would be that having miniforge on the CI hid the install bug, so it may hide another one in the future (I think the average user uses a standard conda instead). That's not very important.

**Pull request closed without merging**